src/common/math/math.h

```diff
@@ -0,0 +1,34 @@
+/**
+ * Math types and utils used in the processor project
+ */
+#ifndef COMMON_MATH_H
+#define COMMON_MATH_H
+
+/**
+ * 3D vector.
+ */
+typedef union {
+    float array[3];
+
+    struct {
+        float x;
+        float y;
+        float z;
+    } axis;
```

Comment:
My gut says `component` might be clearer

Reply:
I copy pasted this directly from Fusion lib 😭
I'm also a fan of component though

Reply:
updated to component